test/core/extensions/crypto_extension_test.cpp

```diff
@@ -422,3 +502,176 @@ TEST_F(CryptoExtensionTest, DISABLED_Secp256k1RecoverCompressed) {
       crypto_ext_->ext_crypto_secp256k1_ecdsa_recover_compressed_v1(sig, msg);
   ASSERT_EQ(ptrsize, res);
 }
+
+/**
+ * @given initialized crypto extensions
+ * @and a damaged secp256k1 signature and message
+ * @when call recovery public secp256k1 compressed key
+ * @then error code is returned
+ */
+TEST_F(CryptoExtensionTest, Secp256k1RecoverCompressedFailure) {
+  WasmPointer sig = 1;
+  WasmPointer msg = 10;
+  WasmSpan res = WasmResult(20, 20).combine();
+
+  auto &sig_input = secp_signature;
+  auto &msg_input = secp_message_hash;
+  Buffer sig_buffer(sig_input);
+  sig_buffer[4] = 0;  // damage signature
```

Comment:
```suggestion
  sig_buffer[4] = ~sig_buffer[4];  // damage signature
```
As above

Reply:
I find zeroizing more appropriate and it works as it should, since I use predefined signature buffer.